### PR TITLE
Enable nullability in RelatedPropertyManager

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in RelatedPropertyManager.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9444)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs [hunk 1]
```diff
@@ -2,31 +2,30 @@
...
     private void Bind(BindingManagerBase parentManager, string dataField)
     {
         Debug.Assert(parentManager is not null, "How could this be a null parentManager.");
-        this.parentManager = parentManager;
-        this.dataField = dataField;
-        fieldInfo = parentManager.GetItemProperties().Find(dataField, true);
-        if (fieldInfo is null)
+        _parentManager = parentManager;
+        _fieldInfo = parentManager.GetItemProperties().Find(dataField, true)!;
+        if (_fieldInfo is null)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs:28]: May be changing to throw right away looks cleaner.
```suggestion
        _fieldInfo = parentManager.GetItemProperties().Find(dataField, true) ?? throw new ArgumentException(string.Format(SR.RelatedListManagerChild, dataField));
```

File: src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs [hunk 2]
```diff
@@ -46,13 +45,13 @@ internal override string GetListName()
         return base.GetListName();
     }
 
-    protected internal override string GetListName(ArrayList listAccessors)
+    protected internal override string GetListName(ArrayList? listAccessors)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs:48]: can you check call sites if any one expecting null reference exception?  or returning string.empty is an option when input is null?

**Review by dreddy-work — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs:45]: The only call site I see is the `GetListName()` above. So, I added a null check.

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/RelatedPropertyManager.cs:45]: returning empty make sense to me.

**Review by dreddy-work — APPROVED** [review]
LGTM.

**Pull request merged**